src/components/calcite-picker-item/calcite-picker-item.tsx

```diff
@@ -0,0 +1,180 @@
+
+  @Prop({ reflect: true }) textHeading: string;
+
+  @Prop({ reflect: true }) textDescription: string;
+
+  @Prop({ reflect: true }) value: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Element() el: HTMLElement;
+
+  @State() isSelected = this.selected;
+
+  dir: "rtl" | "ltr";
+
+  // --------------------------------------------------------------------------
+  //
+  //  Lifecycle
+  //
+  // --------------------------------------------------------------------------
+
+  connectedCallback() {
+    this.dir = getElementDir(this.el);
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  @Event() calcitePickerItemSelectedChange: EventEmitter;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Public Methods
+  //
+  // --------------------------------------------------------------------------
+
+  @Method() async toggleSelected(coerce?: boolean, emit = false) {
+    this.isSelected = typeof coerce === "boolean" ? coerce : !this.isSelected;
+    if (emit) {
+      this.emitChangeEvent();
+    }
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  // --------------------------------------------------------------------------
+
+  pickerClickHandler = (event: MouseEvent): void => {
+    this.isSelected = !this.isSelected;
+    this.emitChangeEvent(event.shiftKey);
+  };
+
+  secondaryActionContainerClickHandler(event: MouseEvent) {
+    event.stopPropagation();
+  }
+
+  emitChangeEvent(shiftPressed = false) {
+    this.calcitePickerItemSelectedChange.emit({
+      item: this.el,
+      value: this.value,
+      selected: this.isSelected,
+      shiftPressed
+    });
+  }
+
+  // --------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  // --------------------------------------------------------------------------
+
+  renderIcon() {
+    const { icon } = this;
+    if (!icon) {
+      return null;
+    }
+    if (icon === ICON_TYPES.grip) {
+      return (
+        <span class={CSS.handle}>
+          <CalciteIcon size="24" path={handleVertical24} />
+        </span>
+      );
+    } else {
+      const path =
+        icon === ICON_TYPES.square
+          ? this.isSelected
+            ? checkSquare16
+            : square16
+          : this.isSelected
+          ? circleFilled16F
+          : circle16F;
+      return (
+        <span class="icon">
+          <CalciteIcon size="16" path={path} />
+        </span>
+      );
+    }
+  }
+
+  render() {
+    const { icon } = this;
+    return (
+      <Host
+        dir={this.dir}
```

Comment:
In other components, we've been using a utility class to set the RTL CSS. The problem with this is that if a `dir` changes on a parent, this will be set on the host.

Reply:
ok I'll do that.

Reply:
worth noting that if dir changes on a parent - this component will likely need to re-render somehow anyway since a parent change wouldn't trigger a re-render.